component/module/git/git.go

```diff
@@ -28,43 +24,30 @@ func init() {
 	})
 }
 
-// Arguments configures the module.git component.
+// Arguments holds values which are used to configure the module.git component.
 type Arguments struct {
-	Repository    string        `river:"repository,attr"`
-	Revision      string        `river:"revision,attr,optional"`
-	Path          string        `river:"path,attr"`
-	PullFrequency time.Duration `river:"pull_frequency,attr,optional"`
-
-	Arguments     map[string]any    `river:"arguments,block,optional"`
-	GitAuthConfig vcs.GitAuthConfig `river:",squash"`
-}
+	RemoteGitArguments remote_git.Arguments `river:",squash"`
 
-// DefaultArguments holds default settings for Arguments.
-var DefaultArguments = Arguments{
-	Revision:      "HEAD",
-	PullFrequency: time.Minute,
+	Arguments map[string]any `river:"arguments,block,optional"`
 }
 
 // SetToDefault implements river.Defaulter.
 func (args *Arguments) SetToDefault() {
-	*args = DefaultArguments
+	args.RemoteGitArguments.SetToDefault()
 }
 
 // Component implements the module.git component.
 type Component struct {
 	opts component.Options
-	log  log.Logger
 	mod  *module.ModuleComponent
 
-	mut      sync.RWMutex
-	repo     *vcs.GitRepo
-	repoOpts vcs.GitRepoOptions
-	args     Arguments
+	mut     sync.RWMutex
+	args    Arguments
+	content string
 
-	argsChanged chan struct{}
-
-	healthMut sync.RWMutex
-	health    component.Health
+	managedRemoteGit *remote_git.Component
+	inUpdate         atomic.Bool
+	isCreated        atomic.Bool
```

Comment:
I was missing the `defer c.isCreated.Store(true)`, I took the same pattern as in the HTTP remote to keep it consistent. I'm not looking for a refactoring because this code should be soon deprecated